libs/protos/protos/steps/sp_steps_detective.proto

```diff
@@ -86,3 +86,22 @@ message DetectiveStep {
   optional bool negate = 3;
   DetectiveType type = 4;
 }
+
+message DetectiveStepResultMatch {
+  DetectiveType type = 1;
+
+  // For JSON payloads, the path to the match
+  string path = 2;
+
+  // For string payloads, the start and end characters of the match
+  // Placeholder for now, will implement in the future
+  //int32 char_index_start = 3;
+  //int32 char_index_end = 4;
+
+  // Value of the match
+  bytes value = 5;
```

Comment:
Does this have to be set? If detective is asked to do "find email PII", this would then just need the path and the next step would operate off of the payload specified in the input_step. Or is there a case where `value` should be used from here?

Reply:
For the purposes of detective/transform no, but more than likely we will need it for say finding values in a payload and setting them in KV

Reply:
Gotcha - as a message/data passing mechanism.